src/system/ExtensionsModule/Api/AdminApi.php

```diff
@@ -1465,4 +1452,21 @@ private function getInstaller($module, array $modInfo)
 
         return $installer;
     }
+
+    /**
+     * Determines whether the current site is the main site within a Multisites system.
+     *
+     * @return boolean True if this is the main site, false otherwise.
+     */
+    private function isMainSite()
+    {
+        $msConfig = $this->serviceManager['multisites'];
+
+        return (
+            ($msConfig['mainsiteurl'] == $this->request->query->get('sitedns', null)
+                && $msConfig['based_on_domains'] == 0)
+            || ($msConfig['mainsiteurl'] == $_SERVER['HTTP_HOST']
+                && $msConfig['based_on_domains'] == 1)
+        );
+    }
```

Comment:
This entire `AdminApi` is deprecated and not used by the new Api. IMO it is pointless to change it unless it is preventing functionality (which it should not be). That is why it was left alone. Everything is happening in `ModuleController`, `BundleSyncHelper`, `ExtensionHelper` and `ExtensionStateHelper` now. That is why I have implemented the listeners.